metadata_service/proxy/atlas_proxy.py

```diff
@@ -736,6 +739,31 @@ def _get_readers(self, qualified_name: str, top: Optional[int] = 15) -> List[Rea
 
         return results
 
+    def _get_programmatic_descriptions(self, parameters: dict) -> List[ProgrammaticDescription]:
+        _programmatic_descriptions = {}
+        programmatic_descriptions: List[ProgrammaticDescription] = []
+
+        for source, text in parameters.items():
```

Comment:
Two things here:
- Check if `app.config['PROGRAMMATIC_DESCRIPTIONS_EXCLUDE_FILTERS']` exists, before the loop, otherwise you are looping every time. 
- You don't need two separate variables to do this operation. i.e., within the same first loop, create the `programmatic_descriptions.append(ProgrammaticDescription(source=source, text=text))` and it will work. 

And if you want to sort then, simply apply sorting afterwards.

Reply:
done!